common.js

```diff
@@ -403,3 +403,12 @@ export function setupPath(newPathEntries) {
   core.addPath(newPath.join(path.delimiter))
   return msys2Type
 }
+
+export function setupJavaHome() {
+  core.startGroup(`Modifying JAVA_HOME for JRuby`)
+  let newHomeVar = `JAVA_HOME_21_${os.arch().toUpperCase()}`;
```

Comment:
According to https://github.com/actions/runner-images/blob/main/images/ubuntu/Ubuntu2204-Readme.md#java it's JAVA_HOME_21_arm64 on macOS arm64 and indeed it is:
https://github.com/eregon/actions-shell/actions/runs/13798182243/job/38594765321

And seeing https://github.com/eregon/actions-shell/actions/runs/13798182243/job/38594765318#step:4:11 it's `JAVA_HOME_21_X64` on linux-aarch64, that makes little sense but it is what it is.

Reply:
I looked at the results, it's always `JAVA_HOME_21_X64` except on macOS-arm64 where it's `JAVA_HOME_21_arm64` instead.

Reply:
Does not appear there are any CI jobs for Linux on AArch64. I caught the bug in "arm64" vs "ARM64" and fixed it for macos. I guess it will also need to check if it's macos and only use the arm64 version there.